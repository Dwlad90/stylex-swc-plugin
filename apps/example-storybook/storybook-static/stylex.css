
@layer priority1, priority2, priority3, priority4;
@layer priority1{
.padding-x1bg5miv{padding:12px 24px}
.padding-x1tamke2{padding:16px}
.padding-x1arfzav{padding:6px 12px}
.padding-x1ff1495{padding:8px 16px}
}
@layer priority2{
.borderColor-x7kbyhx{border-color:#ddd}
.borderColor-x159w2ql{border-color:#e0e0e0}
.borderColor-x9r1u3d{border-color:transparent}
.borderRadius-x1kogg8i{border-radius:6px}
.borderRadius-xur7f20{border-radius:8px}
.borderStyle-xng3xce{border-style:none}
.borderStyle-x1y0btm7{border-style:solid}
.borderWidth-xc342km{border-width:0}
.borderWidth-xmkeg23{border-width:1px}
.textDecoration-x1hl2dhg{text-decoration:none}
.transition-xurpv0g{transition:all.2s ease}
}
@layer priority3{
.alignItems-x6s0dn4{align-items:center}
.backgroundColor-x11jye9f{background-color:#06c}
.backgroundColor-x1596o1h{background-color:#dc3545}
.backgroundColor-x6cgz0f{background-color:#f5f5f5}
.backgroundColor-x12peec7{background-color:white}
.boxShadow-x1mh2kpm{box-shadow:0 2px 4px rgba(0,0,0,.1)}
.boxShadow-xz3rqps{box-shadow:0 4px 8px rgba(0,0,0,.15)}
.boxShadow-x1gnnqk1{box-shadow:none}
.color-x1525slw{color:#333}
.color-x33gn0b{color:#666}
.color-x1awj2ng{color:white}
.cursor-x1ypdohk{cursor:pointer}
.display-x3nfvp2{display:inline-flex}
.fontFamily-xmcqa5u{font-family:-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,sans-serif}
.fontSize-xfifm61{font-size:12px}
.fontSize-xif65rj{font-size:14px}
.fontSize-x1j61zf2{font-size:16px}
.fontSize-xosj86m{font-size:18px}
.fontWeight-xk50ysn{font-weight:500}
.fontWeight-x117nqv4{font-weight:bold}
.justifyContent-xl56j7k{justify-content:center}
.lineHeight-x1evy7pa{line-height:1.5}
.transform-xnn1q72{transform:translateY(0)}
.backgroundColor-xkdfp31:hover{background-color:#0052a3}
.backgroundColor-x37qbrc:hover{background-color:#c82333}
.backgroundColor-x1jmbrsh:hover{background-color:#e8e8e8}
.boxShadow-x1yrdse8:hover{box-shadow:0 4px 8px rgba(0,0,0,.12)}
.transform-x14ow2ja:hover{transform:translateY(-1px)}
.transform-x3kbueh:active{transform:translateY(0)}
}
@layer priority4{
.marginBottom-x1e56ztr{margin-bottom:8px}
.maxWidth-xw5ewwj{max-width:300px}
.minHeight-xe0p6wg{min-height:28px}
.minHeight-xu0aao5{min-height:36px}
.minHeight-x1gg8mnh{min-height:44px}
}