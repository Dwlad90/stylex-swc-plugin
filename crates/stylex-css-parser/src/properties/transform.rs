/*!
CSS Transform property parsing.

Handles transform property syntax with multiple transform functions.
*/

use crate::{css_types::transform_function::TransformFunction, token_parser::TokenParser};
use std::fmt::{self, Display};

/// CSS transform property value
#[derive(Debug, Clone, PartialEq)]
pub struct Transform {
  pub value: Vec<TransformFunction>,
}

impl Transform {
  /// Create a new Transform
  pub fn new(value: Vec<TransformFunction>) -> Self {
    Self { value }
  }

  pub fn parser() -> TokenParser<Transform> {
    use crate::token_parser::tokens;

    TokenParser::one_or_more_separated_by(TransformFunction::parse(), tokens::whitespace())
      .map(Transform::new, Some("to_transform"))
  }
}

impl Display for Transform {
  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
    let function_strings: Vec<String> = self.value.iter().map(|func| func.to_string()).collect();
    write!(f, "{}", function_strings.join(" "))
  }
}

#[cfg(test)]
mod tests {
  use super::*;
  use crate::css_types::{transform_function::*, Angle, Length, LengthPercentage, Percentage};

  #[test]
  fn test_transform_function_creation() {
    let translate = TransformFunction::TranslateAxis(TranslateAxis::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Axis::X,
    ));

    // Test that it can be created without issues
    assert!(matches!(translate, TransformFunction::TranslateAxis(_)));
  }

  #[test]
  fn test_transform_function_display() {
    let translate_x = TransformFunction::TranslateAxis(TranslateAxis::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Axis::X,
    ));
    assert_eq!(translate_x.to_string(), "translateX(10px)");

    let translate = TransformFunction::Translate(Translate::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Some(LengthPercentage::Length(Length::new(
        20.0,
        "px".to_string(),
      ))),
    ));
    assert_eq!(translate.to_string(), "translate(10px, 20px)");

    let rotate = TransformFunction::Rotate(Rotate::new(Angle::new(45.0, "deg".to_string())));
    assert_eq!(rotate.to_string(), "rotate(45deg)");
  }

  #[test]
  fn test_transform_creation() {
    let func1 = TransformFunction::TranslateAxis(TranslateAxis::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Axis::X,
    ));
    let func2 = TransformFunction::Rotate(Rotate::new(Angle::new(45.0, "deg".to_string())));

    let transform = Transform::new(vec![func1, func2]);
    assert_eq!(transform.value.len(), 2);
  }

  #[test]
  fn test_transform_display() {
    let func1 = TransformFunction::TranslateAxis(TranslateAxis::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Axis::X,
    ));
    let func2 = TransformFunction::Rotate(Rotate::new(Angle::new(45.0, "deg".to_string())));

    let transform = Transform::new(vec![func1, func2]);
    assert_eq!(transform.to_string(), "translateX(10px) rotate(45deg)");
  }

  #[test]
  fn test_transform_empty() {
    let transform = Transform::new(vec![]);
    assert_eq!(transform.to_string(), "");
  }

  #[test]
  fn test_transform_equality() {
    let func = TransformFunction::ScaleAxis(ScaleAxis::new(1.5, Axis::X));

    let transform1 = Transform::new(vec![func.clone()]);
    let transform2 = Transform::new(vec![func]);

    assert_eq!(transform1, transform2);
  }

  #[test]
  fn test_transform_parser_creation() {
    // Test that the parser can be created without panicking
    let _parser = Transform::parser();
  }

  #[test]
  fn test_transform_single_function() {
    let transform = Transform::new(vec![
      TransformFunction::Translate(Translate::new(
        LengthPercentage::Percentage(Percentage::new(50.0)),
        Some(LengthPercentage::Percentage(Percentage::new(-50.0))),
      )),
      TransformFunction::Rotate(Rotate::new(Angle::new(45.0, "deg".to_string()))),
      TransformFunction::Scale(Scale::new(1.2, Some(1.2))),
    ]);

    assert_eq!(transform.value.len(), 3);
    assert!(transform.to_string().contains("translate(50%, -50%)"));
    assert!(transform.to_string().contains("rotate(45deg)"));
    assert!(transform.to_string().contains("scale(1.2, 1.2)"));
  }

  #[test]
  fn test_transform_complex_combination() {
    let func1 = TransformFunction::Translate(Translate::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Some(LengthPercentage::Length(Length::new(
        20.0,
        "px".to_string(),
      ))),
    ));
    let func2 = TransformFunction::Rotate(Rotate::new(Angle::new(45.0, "deg".to_string())));
    let func3 = TransformFunction::Scale(Scale::new(1.2, Some(1.2)));

    let transform = Transform::new(vec![func1, func2, func3]);
    assert_eq!(transform.value.len(), 3);
    assert_eq!(
      transform.to_string(),
      "translate(10px, 20px) rotate(45deg) scale(1.2, 1.2)"
    );
  }

  #[test]
  fn test_transform_common_functions() {
    // Test various common transform functions
    let translate = TransformFunction::Translate(Translate::new(
      LengthPercentage::Length(Length::new(10.0, "px".to_string())),
      Some(LengthPercentage::Length(Length::new(
        20.0,
        "px".to_string(),
      ))),
    ));

    let rotate = TransformFunction::Rotate(Rotate::new(Angle::new(45.0, "deg".to_string())));

    let scale = TransformFunction::Scale(Scale::new(1.5, Some(1.5)));

    let skew = TransformFunction::Skew(Skew::new(
      Angle::new(10.0, "deg".to_string()),
      Some(Angle::new(20.0, "deg".to_string())),
    ));

    let functions = vec![translate, rotate, scale, skew];
    let transform = Transform::new(functions);

    assert_eq!(transform.value.len(), 4);
    assert!(transform.to_string().contains("translate"));
    assert!(transform.to_string().contains("rotate"));
    assert!(transform.to_string().contains("scale"));
    assert!(transform.to_string().contains("skew"));
  }
}
