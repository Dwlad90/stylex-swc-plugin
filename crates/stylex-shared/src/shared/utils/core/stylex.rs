use crate::{
  shared::{
    enums::data_structures::fn_result::FnResult, utils::ast::convertors::string_to_expression,
  },
  transform::styleq::common::styleq,
};

use super::parse_nullable_style::ResolvedArg;

pub(crate) fn stylex(values: &Vec<ResolvedArg>) -> Option<FnResult> {
  let result = styleq(values);

  Some(FnResult::Stylex(string_to_expression(
    result.class_name.as_str(),
  )))
}
