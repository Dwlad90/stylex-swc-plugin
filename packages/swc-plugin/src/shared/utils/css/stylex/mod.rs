pub mod evaluate;
pub mod evaluate_stylex_create_arg;
pub(crate) mod flatten_raw_style_object;
pub(crate) mod parse_css;
pub(crate) mod split_css_value;
