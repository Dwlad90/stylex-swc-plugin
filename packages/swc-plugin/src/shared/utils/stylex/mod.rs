pub(crate) mod attrs;
pub(crate) mod define_vars_utils;
pub(crate) mod dev_class_name;
pub(crate) mod js_to_expr;
pub(crate) mod make_string_expression;
pub(crate) mod member_expression;
pub(crate) mod parse_nullable_style;
pub(crate) mod props;
pub(crate) mod stylex;
pub(crate) mod stylex_merge;
pub(crate) mod tests;
