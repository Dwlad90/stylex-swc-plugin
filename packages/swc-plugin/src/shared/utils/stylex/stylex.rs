use crate::{
  shared::{enums::FnResult, utils::common::string_to_expression},
  transform::styleq::styleq::styleq,
};

use super::parse_nullable_style::ResolvedArg;

pub(crate) fn stylex(values: &Vec<ResolvedArg>) -> Option<FnResult> {
  let result = styleq(values);

  let class_name = result.class_name.to_string();

  // let props: Vec<PropOrSpread> = vec![];

  // panic!("Implement stylex");

  // if let Some(value) =  {
  // vec![];
  // };
  //  else {
  //     vec![]
// };

  dbg!(&values, &class_name);

  Some(FnResult::Stylex(string_to_expression(class_name).unwrap()))

  // Some(FnResult::Stylex(
  //     object_expression_factory(vec![prop_or_spread_expression_creator(
  //         String::from("className"),
  //         string_to_expression(class_name).unwrap(),
  //     )])
  //     .unwrap(),
  // ))
}

pub(crate) fn gen_condition_permutations(count: usize) -> Vec<Vec<bool>> {
  let mut result = Vec::new();
  for i in 0..2u32.pow(count as u32) {
    let mut combination = Vec::new();
    for j in 0..count {
      combination.push(i & (1 << j) != 0);
    }
    result.push(combination);
  }
  result
}
