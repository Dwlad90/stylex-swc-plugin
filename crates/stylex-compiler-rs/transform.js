var { transform } = require('./index')

console.log(
  JSON.stringify(
    transform(`
  import stylex from '@stylexjs/stylex';
  export const styles = stylex.create({
      default: {
          backgroundColor: 'red',
          color: 'blue',
      }
  });
  `),
    null,
    2,
  ),
)

/**
 * Output:
  {
    "code": "import stylex from '@stylexjs/stylex';\nexport const styles = {\n    default: {\n        backgroundColor: \"xrkmrrc\",\n        color: \"xju2f9n\",\n        $$css: true\n    }\n};\n",
    "metadata": {
      "stylex": {
        "styles": [
          {
            "className": "xrkmrrc",
            "style": {
              "ltr": ".xrkmrrc{background-color:red}",
              "rtl": null
            },
            "priority": 3000
          },
          {
            "className": "xju2f9n",
            "style": {
              "ltr": ".xju2f9n{color:blue}",
              "rtl": null
            },
            "priority": 3000
          }
        ]
      }
    },
    "sourcemap": "{\"version\":3,\"sources\":[\"<anon>\"],\"names\":[],\"mappings\":\"AACE;AACA;;;;;;EAKG\"}"
  }
 */
