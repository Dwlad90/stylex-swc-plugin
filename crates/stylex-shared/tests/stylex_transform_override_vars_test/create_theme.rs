use insta::assert_snapshot;
use stylex_shared::{
  shared::structures::{
    plugin_pass::PluginPass,
    stylex_options::{StyleXOptions, StyleXOptionsParams},
  },
  StyleXTransform,
};
use swc_core::{
  common::FileName,
  ecma::{
    parser::{Syntax, TsSyntax},
    transforms::testing::test,
  },
};

use crate::utils::transform::stringify_js;

fn get_default_opts() -> StyleXOptionsParams {
  StyleXOptionsParams {
    unstable_module_resolution: Some(StyleXOptions::get_haste_module_resolution(None)),
    class_name_prefix: Some("x".to_string()),
    ..StyleXOptionsParams::default()
  }
}

fn transform(input: &str) -> String {
  stringify_js(
    input,
    Syntax::Typescript(TsSyntax {
      tsx: true,
      ..Default::default()
    }),
    |tr| {
      StyleXTransform::new_test_with_pass(
        tr.comments.clone(),
        PluginPass {
          cwd: None,
          filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
        },
        Some(&mut StyleXOptionsParams {
          runtime_injection: Some(false),
          dev: Some(true),
          ..get_default_opts()
        }),
      )
    },
  )
}

static OUTPUT_OF_STYLEX_DEFINE_VARS: &str = r#"
import stylex from 'stylex';
export const buttonTheme = {
    bgColor: "var(--xgck17p)",
    bgColorDisabled: "var(--xpegid5)",
    cornerRadius: "var(--xrqfjmn)",
    fgColor: "var(--x4y59db)",
    __themeName__: "x568ih9"
};
"#;

static CREATE_THEME: &str = r#"{
  bgColor: {
    default: 'green',
    '@media (prefers-color-scheme: dark)': 'lightgreen',
    '@media print': 'transparent',
  },
  bgColorDisabled: {
    default: 'antiquewhite',
    '@media (prefers-color-scheme: dark)': 'floralwhite',
  },
  cornerRadius: { default: '6px' },
  fgColor: 'coral',
}"#;

static CREATE_THEME_WITH_DIFFERENT_ORDER: &str = r#"{
  bgColorDisabled: {
    default: 'antiquewhite',
    '@media (prefers-color-scheme: dark)': 'floralwhite',
  },
  fgColor: { default: 'coral' },
  bgColor: {
    default: 'green',
    '@media print': 'transparent',
    '@media (prefers-color-scheme: dark)': 'lightgreen',
  },
  cornerRadius: '6px',
}"#;

#[test]
fn variables_order_does_not_change_the_class_name_hash() {
  let input_v1 = format!(
    r#"
    {}
  const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  );

  let input_v2 = format!(
    r#"
    {}
    const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME_WITH_DIFFERENT_ORDER
  );

  let output_v1 = transform(input_v1.as_str());
  let output_v2 = transform(input_v2.as_str());

  assert_snapshot!(output_v1);
  assert_snapshot!(output_v2);

  assert_eq!(output_v1, output_v2);
}

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      runtime_injection: Some(false),
      ..get_default_opts()
    })
  ),
  transforms_variables_object,
  format!(
    r#"
    {}
    export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  transforms_variables_object_and_add_stylex_inject_in_dev_mode,
  format!(
    r#"
    {}
    export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      runtime_injection: Some(false),
      unstable_module_resolution: Some(StyleXOptions::get_common_js_module_resolution(None)),
      ..get_default_opts()
    })
  ),
  transforms_variables_object_in_non_haste_env,
  format!(
    r#"
    {}
    export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      unstable_module_resolution: Some(StyleXOptions::get_common_js_module_resolution(None)),
      ..get_default_opts()
    })
  ),
  transforms_variables_object_in_non_haste_dev_env,
  format!(
    r#"
    {}
    export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      runtime_injection: Some(false),
      ..get_default_opts()
    })
  ),
  transforms_multiple_variables_objects_in_a_single_file,
  format!(
    r#"
    {}
    export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    export const buttonThemeNew = stylex.createTheme(buttonTheme, {{
      bgColor: 'skyblue',
      cornerRadius: '8px',
    }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  transforms_multiple_variables_objects_in_a_single_file_in_dev_mode,
  format!(
    r#"
    {}
    export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    export const buttonThemeMonochromatic = stylex.createTheme(
      buttonTheme, {{
        bgColor: 'white',
        bgColorDisabled: 'black',
        cornerRadius: '0px',
      }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  transforms_variables_objects_with_references_to_local_variables,
  format!(
    r#"
      {}
      const RADIUS = 10;
      export const buttonThemePositive = stylex.createTheme(buttonTheme, {{
      bgColor: {{
        default: 'green',
        '@media (prefers-color-scheme: dark)': 'lightgreen',
        '@media print': 'transparent',
      }},
      bgColorDisabled: {{
        default: 'antiquewhite',
        '@media (prefers-color-scheme: dark)': 'floralwhite',
      }},
      cornerRadius: {{ default: RADIUS }},
      fgColor: 'coral',
    }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  allows_references_to_local_variables_with_static_values,
  format!(
    r#"
      {}
      const COLOR = 'coral';
      export const buttonThemePositive = stylex.createTheme(buttonTheme, {{
        bgColor: {{
          default: 'green',
          '@media (prefers-color-scheme: dark)': 'lightgreen',
          '@media print': 'transparent',
        }},
        bgColorDisabled: {{
          default: 'antiquewhite',
          '@media (prefers-color-scheme: dark)': 'floralwhite',
        }},
        cornerRadius: {{ default: '6px' }},
        fgColor: COLOR,
      }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  allows_template_literal_references,
  format!(
    r#"
      {}
      const name = 'light';
      export const buttonThemePositive = stylex.createTheme(buttonTheme, {{
        bgColor: {{
          default: `${{name}}green`,
          '@media (prefers-color-scheme: dark)': 'lightgreen',
          '@media print': 'transparent',
        }},
        bgColorDisabled: {{
          default: 'antiquewhite',
          '@media (prefers-color-scheme: dark)': 'floralwhite',
        }},
        cornerRadius: {{ default: '6px' }},
        fgColor: 'coral',
      }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  allows_pure_complex_expressions,
  format!(
    r#"
      {}
      const RADIUS = 2;
      export const buttonThemePositive = stylex.createTheme(buttonTheme, {{
        bgColor: {{
          default: 'green',
          '@media (prefers-color-scheme: dark)': 'lightgreen',
          '@media print': 'transparent',
        }},
        bgColorDisabled: {{
          default: 'antiquewhite',
          '@media (prefers-color-scheme: dark)': 'floralwhite',
        }},
        cornerRadius: {{ default: RADIUS * 2 }},
        fgColor: 'coral',
      }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/utils/NestedTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      unstable_module_resolution: Some(StyleXOptions::get_common_js_module_resolution(Some(
        "/stylex/packages/".to_string()
      ))),
      ..get_default_opts()
    })
  ),
  transforms_variables_object_in_common_js_with_nested_file_path,
  format!(
    r#"
      {}
      export const buttonThemePositive = stylex.createTheme(buttonTheme, {});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS, CREATE_THEME
  )
  .as_str()
);

test!(
  Syntax::Typescript(TsSyntax {
    tsx: true,
    ..Default::default()
  }),
  |tr| StyleXTransform::new_test_with_pass(
    tr.comments.clone(),
    PluginPass {
      cwd: None,
      filename: FileName::Real("/stylex/packages/TestTheme.stylex.js".into()),
    },
    Some(&mut StyleXOptionsParams {
      dev: Some(true),
      ..get_default_opts()
    })
  ),
  transforms_typed_object_overrides,
  format!(
    r#"
      {}
      export const RADIUS = 2;
      export const buttonThemePositive = stylex.createTheme(buttonTheme, {{
      bgColor: stylex.types.color({{
        default: 'green',
        '@media (prefers-color-scheme: dark)': 'lightgreen',
        '@media print': 'transparent',
      }}),
      bgColorDisabled: stylex.types.color({{
        default: 'antiquewhite',
        '@media (prefers-color-scheme: dark)': 'floralwhite',
      }}),
      cornerRadius: stylex.types.length({{ default: RADIUS * 2 }}),
      fgColor: stylex.types.color('coral'),
    }});
    "#,
    OUTPUT_OF_STYLEX_DEFINE_VARS
  )
  .as_str()
);
