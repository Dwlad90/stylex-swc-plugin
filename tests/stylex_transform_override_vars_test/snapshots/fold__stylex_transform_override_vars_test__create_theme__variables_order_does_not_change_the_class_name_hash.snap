---
source: tests/stylex_transform_override_vars_test/create_theme.rs
expression: output_v1
---
import _inject from "@stylexjs/stylex/lib/stylex-inject";
var _inject2 = _inject;
import stylex from 'stylex';
export const buttonTheme = {
    bgColor: "var(--xgck17p)",
    bgColorDisabled: "var(--xpegid5)",
    cornerRadius: "var(--xrqfjmn)",
    fgColor: "var(--x4y59db)",
    __themeName__: "x568ih9"
};
_inject2(".xtrlmmh{--xgck17p:green;--xpegid5:antiquewhite;--xrqfjmn:6px;--x4y59db:coral;}", 0.5);
_inject2("@media (prefers-color-scheme: dark){.xtrlmmh{--xgck17p:lightgreen;--xpegid5:floralwhite;}}", 0.6);
_inject2("@media print{.xtrlmmh{--xgck17p:transparent;}}", 0.6);
